#!/usr/bin/env python
PACKAGE = "innfos_joint_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("pos_p", double_t, 0, "P parameter for position control", 1.0, 0.0, 1000.0)
gen.add("pos_i", double_t, 0, "I parameter for position control", 0.0, 0.0, 10.0)
gen.add("pos_d", double_t, 0, "D parameter for position control", 0.0, 0.0, 100.0)
gen.add("vel_p", double_t, 0, "P parameter for velocity control", 1.0, 0.0, 1000.0)
gen.add("vel_i", double_t, 0, "I parameter for velocity control", 0.0, 0.0, 10.0)
gen.add("vel_d", double_t, 0, "D parameter for velocity control", 0.0, 0.0, 100.0)
gen.add("pos_loop", bool_t, 1, "Postion loop",  True)
gen.add("vel_loop", bool_t, 1, "Velocity loop",  True)

exit(gen.generate(PACKAGE, "effort_state_controller", "EffortStateControllerPID"))